js/edge.js

```diff
@@ -0,0 +1,812 @@
+      }
+    } else {
+      exe = io.findInPath(EDGEDRIVER_LEGACY_EXE, true);
+    }
+    if (!exe) {
+      throw Error(
+        'The ' + EDGEDRIVER_LEGACY_EXE + ' could not be found on the current PATH. ' +
+        'Please download the latest version of the MicrosoftEdgeDriver from ' +
+        'https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/ and ' +
+        'ensure it can be found on your PATH.');
+    }
+
+    super(exe);
+    this.setLoopback(true);  // Required for chromium edge
+    this.options_.args.push('--jwp'); // Required for legacy edge
+  }
+
+  /**
+   * Sets which port adb is listening to. _The EdgeDriver will connect to adb
+   * if an {@linkplain Options#androidPackage Android session} is requested, but
+   * adb **must** be started beforehand._
+   *
+   * @param {number} port Which port adb is running on.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  setAdbPort(port) {
+    return this.addArguments('--adb-port=' + port);
+  }
+
+  /**
+   * Sets the path of the log file the driver should log to. If a log file is
+   * not specified, the driver will log to stderr.
+   * @param {string} path Path of the log file to use.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  loggingTo(path) {
+    return this.addArguments('--log-path=' + path);
+  }
+
+  /**
+   * Enables verbose logging.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  enableVerboseLogging() {
+    return this.addArguments('--verbose');
+  }
+
+  /**
+   * Sets the number of threads the driver should use to manage HTTP requests.
+   * By default, the driver will use 4 threads.
+   * @param {number} n The number of threads to use.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  setNumHttpThreads(n) {
+    return this.addArguments('--http-threads=' + n);
+  }
+
+  /**
+   * @override
+   */
+  setPath(path) {
+    super.setPath(path);
+    return this.addArguments('--url-base=' + path);
+  }
+}
+
+
+/** @type {remote.DriverService} */
+let defaultService = null;
+
+
+/**
+ * Sets the default service to use for new MicrosoftEdgeDriver instances.
+ * @param {!remote.DriverService} service The service to use.
+ * @throws {Error} If the default service is currently running.
+ */
+function setDefaultService(service) {
+  if (defaultService && defaultService.isRunning()) {
+    throw Error(
+      'The previously configured EdgeDriver service is still running. ' +
+      'You must shut it down before you may adjust its configuration.');
+  }
+  defaultService = service;
+}
+
+
+/**
+ * Returns the default MicrosoftEdgeDriver service. If such a service has
+ * not been configured, one will be constructed using the default configuration
+ * for an MicrosoftEdgeDriver executable found on the system PATH.
+ * @return {!remote.DriverService} The default MicrosoftEdgeDriver service.
+ */
+function getDefaultService() {
+  defaultService = new ServiceBuilder().build();
```

Comment:
This ignores setDefaultService and creates a new service every time it is called.

Reply:
reversed to the original getDefaultService definition and create a new function createServiceFromCapabilities to create service based on options.